theano/gpuarray/opt.py

```diff
@@ -319,7 +319,8 @@ def apply(self, fgraph):
 
             context_name = None
             for i in [mapping[i] for i in node.inputs]:
-                if isinstance(i.type, GpuArrayType):
+                target = getattr(i.tag, 'target', None)
+                if (isinstance(i.type, GpuArrayType) and target != 'cpu'):
```

Comment:
I made a comment with that I think we need to do:

https://github.com/Theano/Theano/pull/4570/files#r67717390

@abergeron should confirm.